client/question/services/code_evaluators/PythonPrereqCheckService.js

```diff
@@ -63,22 +92,27 @@ tie.factory('PythonPrereqCheckService', [
      *    WRONG_LANGUAGE_ERRORS.
      */
     var detectAndGetWrongLanguageType = function(code) {
-      var codeLines = getNonStringLines(code);
+      var rawCodeArray = obscureStringCharacters(code).split('\n');
 
       for (var i = 0; i < WRONG_LANGUAGE_ERRORS.python.length; i++) {
         var error = WRONG_LANGUAGE_ERRORS.python[i];
         var regexp = new RegExp(error.regExString);
 
-        if (error.allowMultiline) {
-          if (code.search(regexp) !== -1) {
-            return error.errorName;
-          }
-        } else {
-          for (var j = 0; j < codeLines.length; j++) {
-            if (codeLines[j].search(regexp) !== -1) {
-              return error.errorName;
-            }
+        // Lookup character location of error
+        var firstErrorCharIndex = code.search(regexp);
+        if (firstErrorCharIndex !== -1) {
+          var errorLine = null;
```

Comment:
Consider calling this firstErrorLineNumber (and the above as firstErrorColumnNumber) for ease of understanding.

Also this should perhaps be initialized to 0 (and incremented every time the next for loop runs)? That way there's this sense of "it's on line 0 and column C", which means "it's on line 1 and column (C - length-of-line-0 - 1)", etc.